website/docs/r/csi_driver.html.markdown

```diff
@@ -0,0 +1,68 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_csi_driver"
+sidebar_current: "docs-kubernetes-resource-csi-driver"
+description: |-
+  The Container Storage Interface (CSI) is a standard for exposing arbitrary block and file storage systems to containerized workloads on Container Orchestration Systems (COs) like Kubernetes.
+---
+
+# kubernetes_csi_driver
+
+The [Container Storage Interface](https://kubernetes-csi.github.io/docs/introduction.html)
+(CSI) is a standard for exposing arbitrary block and file storage systems to containerized workloads on Container 
+Orchestration Systems (COs) like Kubernetes.
+
+## Example Usage
+
+```hcl
+resource "kubernetes_csi_driver" "example" {
+  metadata {
+    name = "terraform-example"
+  }
+
+  spec {
+    attach_required        = true
+    pod_info_on_mount      = true
+    volume_lifecycle_modes = ["Ephemeral"]
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `metadata` - (Required) Standard csi driver's metadata. For more info see [Kubernetes reference](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata)
+* `attach_required` - (Required) Indicates if the CSI volume driver requires an attachment operation.
```

Comment:
The documentation here is missing that these arguments are inside the `spec` attribute - we should create an additional heading for `spec` and list these attributes there.

Reply:
added